src/gpsea/analysis/pscore/_impl.py

```diff
@@ -110,3 +110,178 @@ def score(
+            if t in idd_tids:
+                return idd_tids.get(t)
+        return 0
+
+    def _term_or_descendant(self,
+                            target_tid: str,
+                            observed_term_ids: typing.List[str]):
+        """
+        Args:
+            target_tid: term of interest
+            observed_term_ids: all terms observed in patient
+
+        Returns:
+            1 if the term or any descendant is present in the patient, otherwise 0
+        """
+        for term_id in observed_term_ids:
+            for desc_tid in self._hpo.graph.get_ancestors(term_id, include_source=True):
+                if desc_tid.value == target_tid:
+                    return 1
+        return 0
+
+    def _term_or_descendant_count(self,
+                                  target_tid: str,
+                                  observed_term_ids: typing.List[str]):
+        """
+        Args:
+            target_tid: term of interest
+            observed_term_ids: all terms observed in patient
+
+        Returns:
+            1 if the term or any descendant is present in the patient, otherwise 0
+        """
+        total_count = 0
+        for term_id in observed_term_ids:
+            for desc_tid in self._hpo.graph.get_ancestors(term_id, include_source=True):
+                if desc_tid.value == target_tid:
+                    total_count += 1
+        return total_count
+
+    def _postnatal_growth_score(self, observed_term_ids: typing.List[str]) -> float:
+        """
+        calculate the postnatal growth component of the score
+        Args:
+            observed_term_ids: terms observed in patient
+
+        Returns:
+
+        """
+        microcephaly = 'HP:0000252'
+        short_stature = 'HP:0004322'
+        macrocephaly = 'HP:0000256'
+        tall_stature = 'HP:0000098'
+        total_count = 0
+        for tid in [microcephaly, short_stature, macrocephaly, tall_stature]:
+            total_count += self._term_or_descendant(tid, observed_term_ids)
+        if total_count > 2:
+            raise ValueError(f"Inconsistent annotations for postnatal growth score {total_count}:  {observed_term_ids}")
+        return total_count
+
+    def _facial_dysmorphism_score(self, observed_term_ids: typing.List[str]) -> float:
+        """
+        This section assigns two points if two or more anomalies are identified in the following
+        categories: hypertelorism, nasal anomalies and ear anomalies. Our implementation counts the total
+        number of terms or descendents of the hypertelorism, Abnormal external nose morphology, and
+        Abnormal pinna morphology.
+
+        Args:
+            observed_term_ids: terms observed in patient
+
+        Returns: facial dysmorphism score (between 0 and 2)
+
+        """
+        hypertelorism = 'HP:0000316'
+        external_nose = 'HP:0010938'
+        pinna_morphology = 'HP:0000377'
+        total_count = len([t for t in observed_term_ids if t == hypertelorism])
```

Comment:
There is a conflict between the pydoc and the actual code - the code actually does not include the Hypertelorism descendants.

I will update the code to include the descendants.

Reply:
Hypertelorism does not have descendents and so this implementation is a bit quicker

Reply:
OK, I'll revert.